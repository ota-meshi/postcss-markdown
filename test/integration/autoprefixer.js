"use strict";

const path = require("path");
const chai = require("chai");
const { jestSnapshotPlugin } = require("mocha-chai-jest-snapshot");
const autoprefixer = require("autoprefixer");
const postcss = require("postcss");
const { listupFixtures } = require("../utils");
const syntax = require("../..");

chai.use(jestSnapshotPlugin());

const FIXTURE_ROOT = path.resolve(
	__dirname,
	"../../test-fixtures/integration/autoprefixer"
);

describe("Integration with autoprefixer", () => {
	for (const { filename, content } of listupFixtures(FIXTURE_ROOT)) {
		describe(`autoprefixer with markdown`, () => {
			it(filename, () =>
				postcss([autoprefixer])
					.process(content, { syntax })
					.then(function (result) {
						const actual = result.content;
						chai.expect(actual).toMatchSnapshot();
					})
			);
		});
	}
});
