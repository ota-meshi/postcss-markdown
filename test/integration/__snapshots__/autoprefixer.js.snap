// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration with autoprefixer autoprefixer with markdown test.md 1`] = `
"# title

\`\`\`css
::-moz-placeholder {
    color: gray;
}
:-ms-input-placeholder {
    color: gray;
}
::placeholder {
    color: gray;
}
\`\`\`
"
`;
